modules/swagger-codegen/src/main/resources/go/api.mustache

```diff
@@ -42,15 +41,27 @@ func (a {{classname}}) {{nickname}} ({{#allParams}}{{paramName}} {{{dataType}}}{
   var httpMethod = "{{httpMethod}}"
  // create path and map variables
   path := a.Configuration.BasePath + "{{path}}"
-{{#pathParams}} path = strings.Replace(path, "{" + "{{baseName}}" + "}", fmt.Sprintf("%v", {{paramName}}), -1)
+{{#pathParams}}  path = strings.Replace(path, "{" + "{{baseName}}" + "}", fmt.Sprintf("%v", {{paramName}}), -1)
 {{/pathParams}}
 
   {{#allParams}}
   {{#required}}
   // verify the required parameter '{{paramName}}' is set
-  if &{{paramName}} == nil {
-      return {{#returnType}}*new({{{returnType}}}), {{/returnType}}*NewAPIResponseWithError("400 - Bad Request"), errors.New("Missing required parameter '{{paramName}}' when calling {{classname}}->{{operationId}}")
+  {{#isListContainer}}
+  if len({{paramName}}) == 0 {
+       return {{#returnType}}*new({{{returnType}}}), {{/returnType}}*NewAPIResponseWithError("400 - Bad Request"), fmt.Errorf("missing required parameter '{{paramName}}' when calling {{classname}}->{{operationId}}")
```

Comment:
In this case, returning a 400 error may mislead the developers by giving the impression that the server did return something (400) and they use tools like tcpdump, wireshark to capture the request and the response (which do not occur), and they will later figure out that there's no request (which is not a good developer experience we want to deliver)

I'll look into other Go SDKs (e.g. go-github, stripe-go) to see how they handle invalid argument exception.